### PR TITLE
Allow status code to be modified via environment variable

### LINKED ISSUE #3: Support other redirect codes
Currently the component always returns 302 Found (temporary redirect).  Should we allow the user to configure it to send 301 Moved Permanently, 307 Temporary Redirect or 308 Permanent Redirect?

### PR DESCRIPTION
**Pull request by karthik2804**
closes #3 

Allow controlling the returned status code using the `STATUS` environment variable.

### CODE REVIEW

File: redirect.go [hunk 1]
```diff
@@ -7,5 +7,10 @@ import (
 
 func main() {
 	dest := os.Getenv("DESTINATION")
-	fmt.Printf("status: 302\nlocation: %s\n\n", dest)
+	status := os.Getenv("STATUS")
```
Comment (endocrimes) [inline redirect.go:10]: Should we require a relevant status, or at least warn+require an override flag if someone is using an "unexpected" redirect status?

Comment (karthik2804) [reply redirect.go:10]: Enforcing valid status codes makes sense to me. 

Just to be sure, in our case the codes would be 

Permanent redirect - 301, 308
Temporary redirect - 302, 303, 307

Correct me if I'm wrong but the special redirects don't apply to this particular use case. 

https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections

File: redirect.go [hunk 2]
```diff
@@ -7,5 +7,20 @@ import (
...
+	if status != "" && is_valid_status(status) {
+		fmt.Printf("status: %s\nlocation: %s\n\n", status, dest)
+	} else {
+		fmt.Printf("status: 302\nlocation1: %s\n\n", dest)
+	}
+}
+
+func is_valid_status(status string) bool {
```
Comment (rajatjindal) [inline redirect.go:18]: nit: use camel case pls

File: redirect.go [hunk 3]
```diff
@@ -7,5 +7,20 @@ import (
...
+	allowed_status := []string{"301", "302", "303", "307", "308"}
```
Comment (rajatjindal) [inline redirect.go:19]: use `http.StatusXXXXX` instead of status codes (if possible)

**Comment by karthik2804** [comment]
#6 supersedes this.

**Pull request closed without merging**